packages/multichain/src/adapters/caip-permission-adapter-permittedChains.ts

```diff
@@ -0,0 +1,136 @@
+import { toHex } from '@metamask/controller-utils';
+import type { Hex } from '@metamask/utils';
+import { KnownCaipNamespace } from '@metamask/utils';
+
+import type { Caip25CaveatValue } from '../caip25Permission';
+import {
+  KnownNotifications,
+  KnownRpcMethods,
+  KnownWalletScopeString,
+} from '../scope/constants';
+import { getUniqueArrayItems, mergeScopes } from '../scope/transform';
+import type { InternalScopesObject, InternalScopeString } from '../scope/types';
+import { parseScopeString } from '../scope/types';
+
+/**
+ * Gets the Ethereum (EIP155 namespaced) chainIDs from the required and optional scopes.
+ * @param caip25CaveatValue - The CAIP-25 caveat value from which to get the Ethereum chainIDs.
+ * @returns An array of Ethereum chainIDs.
+ */
+export const getPermittedEthChainIds = (
+  caip25CaveatValue: Pick<
+    Caip25CaveatValue,
+    'requiredScopes' | 'optionalScopes'
+  >,
+) => {
+  const ethChainIds: Hex[] = [];
+  const sessionScopes = mergeScopes(
+    caip25CaveatValue.requiredScopes,
+    caip25CaveatValue.optionalScopes,
+  );
+
+  Object.keys(sessionScopes).forEach((scopeString) => {
+    const { namespace, reference } = parseScopeString(scopeString);
+    if (namespace === KnownCaipNamespace.Eip155 && reference) {
+      ethChainIds.push(toHex(reference));
+    }
+  });
+
+  return getUniqueArrayItems(ethChainIds);
+};
+
+/**
+ * Adds an Ethereum (EIP155 namespaced) chainID to the required and optional scopes.
+ * @param caip25CaveatValue - The CAIP-25 caveat value to add the Ethereum chainID to.
+ * @param chainId - The Ethereum chainID to add.
+ * @returns The updated CAIP-25 caveat value with the added Ethereum chainID.
+ */
+export const addPermittedEthChainId = (
+  caip25CaveatValue: Caip25CaveatValue,
+  chainId: Hex,
+) => {
+  const scopeString = `eip155:${parseInt(chainId, 16)}`;
+  if (
+    Object.keys(caip25CaveatValue.requiredScopes).includes(scopeString) ||
+    Object.keys(caip25CaveatValue.optionalScopes).includes(scopeString)
+  ) {
+    return caip25CaveatValue;
+  }
+
+  return {
+    ...caip25CaveatValue,
+    optionalScopes: {
+      [KnownWalletScopeString.Eip155]: {
```

Comment:
What is the purpose of this empty EIP-155 scope?

Reply:
In this case I believe this isn't necessary since I believe `setEthAccounts` [handles adding this scope](https://github.com/MetaMask/core/pull/4784#discussion_r1838601089) when needed to park account permissions for Snaps. @jiexi might have to weigh in here in case I'm missing/forgetting something.

Reply:
You're right, Alex. This is redundant and be removed. Thank you both for catching this (and everything else!)

Reply:
Removed here: https://github.com/MetaMask/core/pull/4784/commits/6c38448482c76b96781c0b0a1fa65ac016fb8eed